src/action/auth-mobile.js

```diff
@@ -135,6 +135,55 @@ class AuthAction {
     await this._unlockWallet();
   }
 
+  /**
+   * Initialize the next seed view by setting a new seedIndex or, if all seed
+   * words have been displayed, navigating to the mobile seed verify view.
+   * view.
+   * @return {undefined}
+   */
+  initNextSeedPage() {
+    if (this._store.wallet.seedIndex < 16) {
+      this._store.wallet.seedIndex += 8;
+    } else {
+      this._nav.goSeedVerify();
+    }
+  }
+
+  /**
+   * Initialize the previous seed view by setting a new seedIndex or, if on the
+   * first seed page, navigating to the select seed view.
+   * @return {undefined}
+   */
+  initPrevSeedPage() {
+    if (this._store.wallet.seedIndex >= 8) {
+      this._store.wallet.seedIndex -= 8;
+    } else {
+      this._nav.goSelectSeed();
+    }
+  }
+
+  /**
+   * Verify that the user has written down the generated seed correctly by
+   * checking three random seed words. If the match continue to setting the
+   * wallet pin.
+   * @return {undefined}
+   */
+  checkSeed() {
+    const {
+      wallet: { seedVerify },
+      seedMnemonic,
+      seedVerifyIndexes,
+    } = this._store;
+    if (
+      seedVerify[0] !== seedMnemonic[seedVerifyIndexes[0] - 1] ||
+      seedVerify[1] !== seedMnemonic[seedVerifyIndexes[1] - 1] ||
+      seedVerify[2] !== seedMnemonic[seedVerifyIndexes[2] - 1]
+    ) {
+      this._alert("Seed words don't match", () => {});
+      return;
+    }
+    this.initSetPin();
+  }
```

Comment:
The auth controller is only for pin entry and fingerprint. Seed stuff should go in a different controller. Any reason we can't reuse the logic from the desktop?

Reply:
Fixed!

Reply:
My mistake, fixed! Thought that only referred to `checkSeed` for some reason.